src/2d/shallow/topo_module.f90

```diff
@@ -220,6 +220,22 @@ subroutine read_topo_settings(file_name)
                     topotime(i) = -huge(1.0)
                     call read_topo_file(mxtopo(i),mytopo(i),itopotype(i),topofname(i), &
                         topowork(i0topo(i):i0topo(i)+mtopo(i)-1))
+
+                    ! set topo0save(i) = 1 if this topo file intersects any
+                    ! dtopo file.  This approach to setting topo0save is changed from 
+                    ! v5.4.1, where it only checked if some dtopo point lies within the
+                    ! topo grid, which might not happen for small scale topo
+                    do j=mtopofiles - num_dtopo + 1, mtopofiles
```

Comment:
We assume topo and dtopo arrays both give points where values are specified. I'm working on a PR to fixe the xllcorner problem -- but if this is specified in the header it will just adjust the points that go into the topo arrays, so I think at this point in the code it won't make any difference.